sentry-ruby/spec/sentry/transaction_spec.rb

```diff
@@ -173,6 +173,21 @@
         allow(Sentry.configuration).to receive(:tracing_enabled?).and_return(true)
       end
 
+      context "when event reporting is not enabled" do
+        before do
+          Sentry.configuration.enabled_environments = ["production"]
+        end
+
+        it "sets @sampled to false and return" do
+          transaction = described_class.new(sampled: true, hub: Sentry.get_current_hub)
+          transaction.set_initial_sample_decision(sampling_context: {})
+          expect(transaction.sampled).to eq(false)
```

Comment:
Thought about this for a while. At first I thought neither `true` nor `false` fit here, because both of those mean a sampling decision was actually made, in which case `nil` would be more appropriate.

However, the [sampling rule](https://develop.sentry.dev/sdk/performance/#sampling) is that explicit sample decision always has precedence:

> 1. If a sampling decision is passed to startTransaction (startTransaction({name: "my transaction", sampled: true})), that decision will be used, regardlesss of anything else

So in this particular example, since the sampling decision was passed to `.new`, it should not have changed.

In any case, tracing is effectively disabled, `startTransaction` should not instantiate transactions.

Reply:
I've addressed the decision making issue in https://github.com/getsentry/sentry-ruby/pull/1423

Reply:
Thanks @st0012. I've chatted with the team and looked at code in other SDKs, I retract my comment for now.
Python and JS do exactly as Ruby does now, despite not adhering 100% to the spec as written.
So instead of going back and forth with this, let's just leave it as it is for now, and revisit it later as we rethink tracing as a whole.

Reply:
Yeah, I implemented this behavior based on what I saw in `sentry-python`. I thought the spec you posted was to correct that.